crates/sui-verifier/src/char_type_verifier.rs

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! A module can define a "characteristic type", that is a type that is instantiated only once, and
```

Comment:
Bit of a naming nit, I think we are going with "one-time-witness type" for anything user facing. As it sort of describes what it is supposed to be used for. 

So a general s/characteristic/one-time-witness/

Reply:
With or without the hyphens, doesn't matter too much

Reply:
And open to discussion here :) I just think calling this thing a one-time witness everywhere might help people understand what is going on and why they want to use it (assuming they understand witnesses)

Reply:
To me a characteristic type was a way to implement one-time witness. For now I have updated the comment to reflect that but I also open for a discussion :-)

Reply:
I would agree with that description, but I think it is the _only_ way to implement a one-time witness. At least the only safe way. 
And I've been pushing more for naming in the Sui Framework for things to be named what they are, not how they work. Really, the name should reflect the most important and distinguishing part of the thing. 


That being said, I could see one-time witness being confusing as there are lots of ways to implement one-time witness, though all those other ways would be not as safe. 

Thoughts given this perspective?

Reply:
Fine with landing the naming as is. I will be curious what @sblackshear thinks, as he came up with the initial name